src/CodeGen_C.cpp

```diff
@@ -25,6 +25,7 @@ const string buffer_t_definition =
     "#ifndef BUFFER_T_DEFINED\n"
     "#define BUFFER_T_DEFINED\n"
     "#include <stdint.h>\n"
+    "#pragma pack(push, 1)\n"
```

Comment:
memoization (and possibly other things) is currently broken with 32-bit
builds. To work around the change where clang now makes in a specific
architecture as a function parameter we started compiling things with the
most generic 32-bit target triple we could find - le32-unknown-nacl (there
isn't actually anything nacl specific about it). However, it turns out that
this generic triple pads bools out to 32 bits, and you get a
sizeof(buffer_t) of 72, even when compiling for 32 bits.

Rather than mess with the triple more, I figured we shouldn't even be
sensitive to these sorts of changes, because we should just specify layouts
explicitly.

On Mon, Mar 30, 2015 at 4:51 PM, Zalman Stern <notifications@github.com>
wrote:

> In src/CodeGen_C.cpp
> <https://github.com/halide/Halide/pull/721#discussion_r27443052>:
>
> > @@ -25,6 +25,7 @@ const string buffer_t_definition =
> >      "#ifndef BUFFER_T_DEFINED\n"
> >      "#define BUFFER_T_DEFINED\n"
> >      "#include <stdint.h>\n"
> > +    "#pragma pack(push, 1)\n"
>
> Is the pragma just for safety or is there really a platform that inserts
> extra space in this structure? (I understand that the padding is necessary
> to prevent some compilers from producing a sizeof that is 2 bytes less than
> others.)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/halide/Halide/pull/721/files#r27443052>.
>